src/utils/popups.js

```diff
@@ -133,25 +186,27 @@ const getLatestFlowReadingInfo = async (url) => {
   return `${reading.value} ${readingUnit} - ${reading.date}`;
 };
 
-export const setupLayerPopups = (map) => {
+export const setupLayerPopups = (map, setMapContext) => {
   map.on("click", "hydroNodes", (e) => {
+    if (window.WCLinkSelectMode) return;
     if (e.originalEvent.defaultPrevented) return;
     e.originalEvent.preventDefault();
 
     const coords = getCoords(e);
     const text = hydroNodePropertiesToHTML(e.features[0]);
 
-    newPopup(coords, text, map);
+    newPopup(coords, text, map, setMapContext);
   });
 
   map.on("click", "watercourseLinks", (e) => {
+    if (window.WCLinkSelectMode) return;
     if (e.originalEvent.defaultPrevented) return;
     e.originalEvent.preventDefault();
 
     const coords = e.lngLat;
     const text = watercourseLinkPropertiesToHTML(e.features[0]);
 
-    newPopup(coords, text, map);
+    newPopup(coords, text, map, setMapContext);
   });
 
   map.on("click", "riverFlowSites", async (e) => {
```

Comment:
Not sure how easy it will be, but is it possible to add the new _highlighting nearest watercourse link_ to the `riverFlowSites` too?

There's not loads of these sites, but I search for and use `Denham` cause there's a couple there!
![image](https://user-images.githubusercontent.com/32230328/195140261-91ccbd8f-28d5-42fe-871c-aa0c9d6f86ae.png)

Reply:
it shouldn't a problem at all.. I'll double check that it works as expected for these too. And thanks for the tip to find one! I had a hard time randomly coming across them.